### PR TITLE
e2e_api: stop using deprecated framework.ExpectNotEqual

### PR DESCRIPTION
**Pull request by Rei1010**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
test/e2e/apimachinery stop using deprecated framework.ExpectNotEqual

#### Which issue(s) this PR fixes:

Fixes #115961

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: test/e2e/apimachinery/discovery.go [hunk 1]
```diff
@@ -142,8 +142,8 @@ var _ = SIGDescribe("Discovery", func() {
 			checkGroup := &metav1.APIGroup{}
 			apiPath := "/apis/" + group.Name + "/"
 			err = f.ClientSet.Discovery().RESTClient().Get().AbsPath(apiPath).Do(ctx).Into(checkGroup)
-			framework.ExpectNoError(err, "Fail to access: %s", apiPath)
-			framework.ExpectNotEqual(len(checkGroup.Versions), 0, "No version found for %v", group.Name)
+			framework.ExpectNoError(err, "Fail to access: %s", apiPath))
```
Comment (jiahuif) [inline test/e2e/apimachinery/discovery.go:145]:
```suggestion
			framework.ExpectNoError(err, "Fail to access: %s", apiPath)
```
There is a stray ')' at the end of the line. Please fix it and I'd be happy to lgtm again :+1:

**Comment by Rei1010** [comment]
@jiahuif ptal

**Review by pohly — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**